subgroup_analysis_module.py

```
@@ -101,11 +108,34 @@ def analyze(
+                 )
+                 missing_data_info = {
+                    'strategy': 'complete-case',
+                    'rows_analyzed': impact['final_rows'],
+                    'rows_excluded': impact['rows_removed'],
+                    'summary_before': missing_summary.to_dict('records')
+                 }
+            else:
+                 df_clean = self.df[cols_to_use].dropna().copy()
             
             if len(df_clean) < 10:
                 raise ValueError(f"Insufficient data: {len(df_clean)} rows")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)